### PR TITLE
[release/v2.26] Remove CentOS as supported Operating System

### LINKED ISSUE #13580: CentOS 7 has reached EOL
CentOS 7 reached EOL; we need to evaluate and finalize our stance on CentOS support in KKP, MC, and OSM going forward.

/milestone KKP 2.26
/assign @csengerszabo

### PR DESCRIPTION
**Pull request by julioc-p**
**What this PR does / why we need it**:
This PR removes CentOS as a supported operating system.

**Which issue(s) this PR fixes**:

Fixes #13580 

**What type of PR is this?**

/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
CentOS removed as a supported operating system.
```

**Documentation**:

```documentation
https://github.com/kubermatic/docs/pull/1804
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/retitle [release/v2.26] Remove CentOS as supported Operating System

File: .prow/dualstack.yaml [hunk 1]
```diff
@@ -564,43 +564,9 @@ presubmits:
...
-                  key: serviceAccountSigningKey
-          securityContext:
-            privileged: true
-          resources:
-            requests:
-              memory: 4Gi
-              cpu: 2
+
+ 
+
```
Comment (ahmedwaleedmalik) [inline .prow/dualstack.yaml]: Drop the extra newlines

File: go.mod [hunk 2]
```diff
@@ -89,8 +89,8 @@ require (
 	helm.sh/helm/v3 v3.14.3
 	k8c.io/kubelb v1.1.2
 	k8c.io/kubeone v1.7.3
-	k8c.io/machine-controller v1.60.0
-	k8c.io/operating-system-manager v1.6.0
+	k8c.io/machine-controller v1.60.1-0.20241118073303-4f251c4bc5a3
+	k8c.io/operating-system-manager v1.6.1-0.20241118134103-5db575f65108
```
Comment (ahmedwaleedmalik) [inline go.mod:93]: Just so you know, you'll need to backport your changes to release branches of MC v1.60 and OSM v1.6. New patch releases will be required for this and those versions should be used here.

We can merge these PRs with the current tags but just make sure that we take care of updating to actual semver tags afterwards.

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]
Although it's benign but i'd recommend not updating anything in the `docs/proposals` directory. Those are historic documents and should represent the state in which they were added to the repo.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by ahmedwaleedmalik** [comment]
cc @kubermatic/sig-release-managers

**Comment by xrstf** [comment]
SIG Release is still unsure about this and sadly SIG Cluster last week did not note down the reasons that speak _for_ a backport. We'll bring this up again in the Engineering Management Sync, as we also do not urgently need this backport (next KKP patch releases are planned for 2.26).

**Comment by xrstf** [comment]
We just talked about this again and now it's clear that CentOS is so dead and broken, there is basically no chance of anyone using it in KKP 2.26.

**Pull request merged**